### DISCUSSION TITLE
how to run fyne app on a target machine which is provided as a virtual pc but withou the 3D support, that means no OpenGL available

### DISCUSSION THREAD

**Discussion by xinfengxia** [fyne-io/fyne#4231] [General]
when i try to run a fyne app, but found can not start, because the virtual pc don't has the OpenGL suport. I looked through on net for sollution , but it seems to me need to use other lib that don't OpenGL is not a prerequesit, that means all ui codes need to shift to a new lib that is totally new . If there is other solutions, because the ui is quite simple, even without any png element, common widget like button, list, radio group

**Comment by andydotxyz** [comment]
Please provide more information - like the client and host OS etc. the solution would be platform dependent, though if you can just turn on "3d acceleration" option for virtual machine it should work without extra steps.

**Reply by Ju-B** [reply]
For windows (in cheap azure vms for instance) i got this working: http://mesa.fdossena.com : a dll to drop next to .exe
